helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -558,23 +558,30 @@ protected void refresh(Map<String, Map<String, Map<String, CurrentState>>> curre
 
   private void resetRoutingTableAndNotify(long startTime, RoutingTable newRoutingTable) {
     _routingTableRef.set(newRoutingTable);
-    logger.info("Refresh the RoutingTable for cluster {}, takes {} ms.",
-        (_helixManager != null ? _helixManager.getClusterName() : null),
+    String clusterName = _helixManager != null ? _helixManager.getClusterName() : null;
```

Comment:
Should we print out a different log entirely if helixManager is null and return? Since it signals a bigger problem?

Reply:
The code is always like this, maybe for the reason that some traditional deployment code can indeed have helixManager to be null and people don't want them to fail. I guess the most immediate previous change in the piece of code was from you and it was still kept this way. So let us just keep it this way.